packages/pnpm-sync-lib/src/pnpmSyncCopy.ts

```diff
@@ -93,17 +99,34 @@ export async function pnpmSyncCopyAsync(options: IPnpmSyncCopyOptions): Promise<
 
   const pnpmSyncJsonFolder = path.dirname(pnpmSyncJsonPath);
 
-  //read the .pnpm-sync.json
-  const pnpmSyncJson = JSON.parse(pnpmSyncJsonContents);
+  // read the .pnpm-sync.json
+  const pnpmSyncJson: IPnpmSyncJson = JSON.parse(pnpmSyncJsonContents);
+
+  // verify if the version is outdated
+  const pnpmSyncJsonVersion: string = getPnpmSyncJsonVersion();
+
+  if (pnpmSyncJsonVersion !== pnpmSyncJson.version) {
+    const errorMessage = `The .pnpm-sync.json file in ${pnpmSyncJsonFolder} is outdated; regenerate it and try again.`;
+    logMessageCallback({
+      message: errorMessage,
+      messageKind: LogMessageKind.ERROR,
+      details: {
+        messageIdentifier: LogMessageIdentifier.COPY_ERROR_NO_SYNC_FILE,
```

Comment:
Make a new identifier such as `COPY_ERROR_INCOMPATIBLE_SYNC_FILE`

Reply:
Also it might be helpful for the message event to include the actual and expected version numbers.